miasm2/ir/ir.py

```diff
@@ -176,6 +176,9 @@ def merge_multi_affect(self, affect_list):
             e_colision = reduce(
                 lambda x, y: x + y, [e.get_modified_slice() for e in expr_list])
```

Comment:
This creates two lists (one as the output of `reduce()`, one by list comprehension.

What do you think of replacing lines 176 to 180 by something like:

```python
e_colision = reduce(lambda x, y: set(x).union(y), (e.get_modified_slice() for e in expr_list))
```

If you really need a list use `e_colision = list(reduce([...]))`. And anyway drop the `set()` in `known_intervals = [...]`